File: src/main/java/com/mercadez/service/ContatoService.java
package com.mercadez.service;

import com.mercadez.dto.ContatoRequest;
import com.mercadez.model.Contato;
import com.mercadez.repository.ContatoRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class ContatoService {

    private final ContatoRepository repo;

    public ContatoService(ContatoRepository repo) { this.repo = repo; }

    @Transactional
    public void enviar(ContatoRequest req) {
        Contato c = Contato.builder()
                .nome(req.getNome())
                .email(req.getEmail().toLowerCase())
                .mensagem(req.getMensagem())
                .build();
        repo.save(c);
    }
}

File: app/services/contato_service.py
"""Equivalente Python de service/ContatoService.java"""
from sqlalchemy.orm import Session

from app.models.contato import Contato
from app.schemas.contato import ContatoRequest


def enviar(db: Session, req: ContatoRequest) -> None:
    contato = Contato(
        nome=req.nome,
        email=req.email.lower(),
        mensagem=req.mensagem,
    )
    db.add(contato)
    db.commit()
